google/cloud/bigquery/dataset.py

```diff
@@ -527,13 +527,46 @@ class Dataset(object):
         "friendly_name": "friendlyName",
         "default_encryption_configuration": "defaultEncryptionConfiguration",
         "storage_billing_model": "storageBillingModel",
+        "default_rounding_mode": "defaultRoundingMode",
     }
 
     def __init__(self, dataset_ref) -> None:
         if isinstance(dataset_ref, str):
             dataset_ref = DatasetReference.from_string(dataset_ref)
         self._properties = {"datasetReference": dataset_ref.to_api_repr(), "labels": {}}
 
+    @property
+    def default_rounding_mode(self):
+        """
+        optional[str]: it could have one of the following possible value.
+        ROUNDING_MODE_UNSPECIFIED: Unspecified will default to using ROUND_HALF_AWAY_FROM_ZERO.
+        ROUND_HALF_AWAY_FROM_ZERO: ROUND_HALF_AWAY_FROM_ZERO rounds half values away from zero when applying precision
+                                    and scale upon writing of NUMERIC and BIGNUMERIC values.
+                                    For Scale: 0 1.1, 1.2, 1.3, 1.4 => 1 1.5, 1.6, 1.7, 1.8, 1.9 => 2
+        ROUND_HALF_EVEN: ROUND_HALF_EVEN rounds half values to the nearest even value when applying precision and scale
+                        upon writing of NUMERIC and BIGNUMERIC values.
+                        For Scale: 0 1.1, 1.2, 1.3, 1.4 => 1 1.5 => 2 1.6, 1.7, 1.8, 1.9 => 2 2.5 => 2
+        """
+        return self._properties.get("defaultRoundingMode")
+
+    @default_rounding_mode.setter
```

Comment:
The test coverage is not passing because the setter function is not covered. Could you please also add unit test for this? Thank you!

Reply:
@Linchin  should I modify this test_create_dataset_w_attrs ?

Reply:
I think it would be better to add several smaller tests (in parallel to `test_create_dataset_with_default_rounding_mode()` as you added), each for a case in our if conditions. These include:
1. `value` is `None`
1. `value` is not a string
2. `value` not in `possible_values`
3. `value` is in `possible_values`